suricata/update/commands/updatesources.py

```diff
@@ -27,11 +28,36 @@
 
 logger = logging.getLogger()
 
+
 def register(parser):
     parser.set_defaults(func=update_sources)
 
+
+def compare_sources(initial_content, final_content):
+    initial_sources = initial_content.get("sources")
+    final_sources = final_content.get("sources")
+    added_sources = {source: final_sources[source]
```

Comment:
Just a minor improvement here:

How about we take the extra items beforehand like:

```removed_sources = dict(set(initial_sources.items()) - set(final_sources.items()))```

and

```added_sources = dict(set(final_resources.items()) - set(initial_sources.items()))```

IMHO, it would be more readable and pythonic.

Reply:
Yes, this looks so much better. Thanks  a lot!

Reply:
Turns out this does not work well for nested dictionaries. From all things I can find, I would end up processing more. Do you have some suggestion that works well for nested dictionaries for Python 2 as well as Python 3?

Reply:
Hey, on giving a second thought to my suggestion, I realized it won't have the same result that you are expecting here, because set difference right now will compare both key and value in the tuples. In order to get the same result you would have to also add something like:

```extra_items = set(initial_sources.keys()) - set(final_resources.keys())
{k:v for k, v in initial_sources.items() if k in extra_items}```

which isn't making much sense anymore, given the already written solution.

I think just iterating like `{k: v for k, v in whatever_source}` will make it clear that the dict is just being filtered, or just using filter with lambda or just doing the following:
`{ k : second_dict[k] for k in set(second_dict) - set(first_dict) }`